src/core/services/interimElement/interimElement.js

```diff
@@ -310,10 +310,14 @@ function InterimElementProvider() {
 
           return interimElement
             .show()
-            .catch(function(reason) { return reason; })
+            .then(function () {
+              showingInterims.push(interimElement);
+            })
+            .catch(function (reason) {
+              return reason;
+            })
```

Comment:
Can you please add a unit test which verifies this fix?

Reply:
That's strange, I'm not sure what happened but I didn't get notified about this comment until recently (when you posted it 3 days ago).

I don't think that your unit test needs to be quite so intrusive. Maybe you can test it via dialogs like in the reproduction for https://github.com/angular/material/issues/11460?